network_analysis/find-service-areas-for-multiple-facilities/src/main/java/com/esri/samples/find_service_areas_for_multiple_facilities/FindServiceAreasForMultipleFacilitiesSample.java

```diff
@@ -127,24 +126,13 @@ public void start(Stage stage) {
         // zoom to the extent of the feature layer
         mapView.setViewpointGeometryAsync(facilitiesFeatureLayer.getFullExtent(), 130);
 
-        // enable the find service areas button when the draw status is completed for the first time
-        mapView.addDrawStatusChangedListener(new DrawStatusChangedListener() {
-          @Override
-          public void drawStatusChanged(DrawStatusChangedEvent drawStatusChangedEvent) {
-            if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.COMPLETED) {
-              // enable the 'find service areas' button
-              findServiceAreasButton.setDisable(false);
-              mapView.removeDrawStatusChangedListener(this);
-            }
-          }
-        });
+        // disable the find service areas button when the feature layer has not loaded
+        findServiceAreasButton.disableProperty().bind(Bindings.createBooleanBinding(()->
+          facilitiesFeatureLayer.getLoadStatus().equals(LoadStatus.NOT_LOADED)));
```

Comment:
This will enable the button if the layer fails to load. Better to check for not equals `LoadStatus.LOADED`?

Also, keep in mind the `loadStatus` property might get into the release, in which case this could change to use it.

Reply:
I have made the changes and tested the sample again but not sure what's causing the task failure on your side.